ebs/ebs.go

```diff
@@ -293,6 +303,199 @@ func (d *Driver) getTypeAndIOPS(opts map[string]string) (string, int64, error) {
 	return volumeType, iops, nil
 }
 
+func convertEc2TagsToMap(tags []*ec2.Tag) map[string]string {
+	tagMap := make(map[string]string)
+	for _, tag := range tags {
+		tagMap[*tag.Key] = *tag.Value
+	}
+	return tagMap
+}
+
+func (d *Driver) MarkVolumeForGC(volume *ec2.Volume) error {
+	if volume == nil {
+		return nil
+	}
+	gcTag := make(map[string]string)
+	gcTag["GarbageCollection"] = time.Now().String()
+	if err := d.ebsService.AddTags(*volume.VolumeId, gcTag); err != nil {
+		return err
+	}
+	return nil
+}
+
+func (d *Driver) CreateAndBuildFromSnapshot(oldVolume *ec2.Volume, snapshotVolume *ec2.Volume, opts map[string]string) (string, int64, error) {
+	// If the volume I would snapshot off of is the same as the one that I need to create then just return
+	if oldVolume != nil && *oldVolume.VolumeId == *snapshotVolume.VolumeId {
+		return *oldVolume.VolumeId, *oldVolume.Size * GB, nil
+	}
+	csr := &CreateSnapshotRequest {
+		VolumeID: *snapshotVolume.VolumeId,
+		Description: fmt.Sprintf("Creating snaphot from %s", *snapshotVolume.VolumeId),
+		Tags: convertEc2TagsToMap(snapshotVolume.Tags),
+	}
+	// Create snapshot from volume in other AZ
+	snapshotId, err := d.ebsService.CreateSnapshot(csr)
+	if err != nil {
+		return "", -1, err
+	}
```

Comment:
Do you think we should wait for the snapshot status to be completed before we submit a request to from it?